common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -277,8 +283,74 @@ public Object run() {
             }
 
             UNALIGNED = unaligned;
+
+            Object maybeException = AccessController.doPrivileged(new PrivilegedAction<Object>() {
+                @Override
+                public Object run() {
+                    try {
+                        // Java9 has jdk.internal.misc.Unsafe and not all methods are propergated to
+                        // sun.misc.Unsafe
+                        Class<?> internalUnsafeClass = getClassLoader(PlatformDependent0.class)
+                                .loadClass("jdk.internal.misc.Unsafe");
+                        Method method = internalUnsafeClass.getDeclaredMethod("getUnsafe");
+                        return method.invoke(null);
+                    } catch (ClassNotFoundException e) {
+                        return e;
+                    } catch (IllegalAccessException e) {
+                        return e;
+                    } catch (InvocationTargetException e) {
+                        return e;
+                    } catch (NoSuchMethodException e) {
+                        return e;
+                    } catch (SecurityException e) {
+                        return e;
+                    }
+                }
+            });
+
+            if (!(maybeException instanceof Throwable)) {
+                internalUnsafe = maybeException;
+                final Object finalInternalUnsafe = internalUnsafe;
+                maybeException = AccessController.doPrivileged(new PrivilegedAction<Object>() {
+                    @Override
+                    public Object run() {
+                        try {
+                            return finalInternalUnsafe.getClass().getDeclaredMethod(
+                                    "allocateUninitializedArray", Class.class, int.class);
+                        } catch (NoSuchMethodException e) {
+                            return e;
+                        } catch (SecurityException e) {
+                            return e;
+                        }
+                    }
+                });
+
+                if (maybeException instanceof Method) {
+                    try {
+                        allocateArrayMethod = (Method) maybeException;
+                        byte[] bytes = (byte[]) allocateArrayMethod.invoke(finalInternalUnsafe, byte.class, 8);
+                        assert bytes.length == 8;
+                    } catch (IllegalAccessException e) {
+                        allocateArrayMethod = null;
+                        maybeException = e;
+                    } catch (InvocationTargetException e) {
+                        allocateArrayMethod = null;
+                        maybeException = e;
+                    }
+                } else {
+                    allocateArrayMethod = null;
```

Comment:
nit: initialize allocateArrayMethod to `null` above so you don't need this else?